lib/src/app/data.dart

```diff
@@ -103,6 +163,54 @@ class DataClient {
     return response.rawData.map((e) => BsonCodec.deserialize(BsonBinary.from(e))).toList();
   }
 
+  /// Obtain unified tabular data and metadata from the specified data source.
+  ///
+  /// Returns a stream of data points.
+  ///
+  /// For more information, see [Data Client API](https://docs.viam.com/appendix/apis/data-client/).
+  Stream<TabularDataPoint> exportTabularData(
+    String partId,
+    String resourceName,
+    String resourceSubtype,
+    String methodName,
+    DateTime? startTime,
+    DateTime? endTime,
+  ) async* {
+    final interval = CaptureInterval();
+    if (startTime != null) {
+      interval.start = Timestamp()..seconds = Int64((startTime.millisecondsSinceEpoch / 1000).floor());
+    }
+    if (endTime != null) {
+      interval.end = Timestamp()..seconds = Int64((endTime.millisecondsSinceEpoch / 1000).floor());
+    }
+
+    final request = ExportTabularDataRequest()
+      ..partId = partId
+      ..resourceName = resourceName
+      ..resourceSubtype = resourceSubtype
+      ..methodName = methodName
+      ..interval = interval;
+    final responses = _dataClient.exportTabularData(request);
```

Comment:
I believe this returns a stream, and you might be able to map this directly, instead of having to iterate and yield like you do below